lib/config.cpp

```diff
@@ -823,7 +831,8 @@ std::optional<Mapping> Config::get_3_tier_model_mapping(const std::string& modul
     }
     auto& mapping = module_tier_mappings.value();
     if (mapping.implementations.find(impl_id) == mapping.implementations.end()) {
-        return std::nullopt;
+        // if no specific implementation mapping is given, use the module mapping
+        return mapping.module;
     }
     return mapping.implementations.at(impl_id);
```

Comment:
This pattern happens quite often.  It is usually better to use the iterator instead of doing the lookup twice:
```
const auto mapping_it = ...find(impl_id);
if (mapping_id == std::end(...)) {
  return default;
}
return *mapping_it;
```

Reply:
I guess something like this would be quite readable:
```cpp
const auto& mappings = module_tier_mappings.value();
const auto mapping_it = mappings.implementations.find(impl_id);
if (mapping_it == mappings.implementations.end()) {
    return mappings.module;
}
const auto& [key, mapping] = (*mapping_it);
return mapping;
```

The alternative doesn't quite feel as readable:
```cpp
...
return (*mapping_it).second;
```
Or what do you think?